runners/samza/src/test/java/org/apache/beam/runners/samza/util/RunWithTimeoutTest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.runners.samza.util;
+
+import java.util.concurrent.TimeoutException;
+import org.junit.Test;
+
+public class RunWithTimeoutTest {
```

Comment:
Add another test to capture the scenario where runnable throws an exception (mimicking the user code throwing an exception) and ensure if it is bubbled up?

Looks like the code still doesn't bubble up the original cause but throws `ExecutionException` instead.

Reply:
Added a new test, can you please check if that is what you wanted to verify?